### PR TITLE
Code block: Try an experimentalSelector

### LINKED ISSUE #31135: theme.json: Code block typography fontFamily isn't working
Here's the markup for the code block:

```
<pre class="wp-block-code"><code></code></pre>
```

When setting a fontFamily:

```
"core/code": {
	"typography": {
		"fontFamily": "var(--wp--preset--font-family--base)"
	}
}
```
 
The style gets applied to `.wp-block-code`, but that's the `<pre>` tag, and this causes the **user agent stylesheet**
 to take precedence:

code

<hr>

The solution here would be to apply `font-family` to `.wp-block-code code` rather than `.wp-block-code`.

cc: https://github.com/Automattic/themes/issues/3640

### PR DESCRIPTION
**Pull request by carolinan**
## Description

Fixes https://github.com/WordPress/gutenberg/issues/31135
Styles are applied to `.wp-block-code > code` to override the user agent stylesheet.

## How has this been tested?

Tested manually by:
Add the sample typography code from the issue to theme.json of TT1 Blocks, or activate the theme <del>Quadrant</del> Excuse me: Quadrat.
Add a code block in either editor
View source to confirm that the correct font is used.

## Screenshots 

Quadrat, after:
code-block-styles-quadrat

TT1 Blocks before (top) and after (bottom).

Because the block is selected, the border is not as visible.
The difference is clear in this theme because the theme adds padding to the code block.
In the after image, the font verdana is used.

code-block-styles

Empty theme, after:
code-block-styles-empty-theme

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by scruffian — APPROVED** [review]
Thanks for working on this - works as advertised.

File: packages/block-library/src/code/block.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
 	},
 	"supports": {
 		"anchor": true,
+		"__experimentalSelector": ".wp-block-code > code",
```
Comment (youknowriad) [inline packages/block-library/src/code/block.json:17]: The issue with this is that the styles from global styles applies to the inner "code" but not the styles from the "fontSize" block support and any block support we add here in the future.

Comment (aristath) [reply packages/block-library/src/code/block.json:17]: Good point, should we revert this PR until we figure out an alternative?

**Comment by ndiego** [comment]
I did want to note that while this PR does fix the font-family issue, any additional styles added via theme.json, such a background color, border-radius etc are added to the inner `code` when they should be applied to the outer `pre`. Looks like @youknowriad mentioned this above as well. 

One possible solution is to just change the theme.scss file so the theme styles target `.wp-block-code > code` instead of just `.wp-block-code`. Then any global styles would override the theme styles as expected. Not sure this is the best approach but the only one I can think of at the moment that would theoretically work. Might have broader implications though...

**Pull request merged**